### PR TITLE
104061+104063 reject and unclear endpoint

### PR DESCRIPTION
**Pull request by janid1967**
[AB#104061](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/104061) 
[AB#104063](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/104063)

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.WebApi/Controllers/PunchItems/PunchItemsController.cs [hunk 1]
```diff
@@ -117,6 +118,21 @@ public class PunchItemsController : ControllerBase
         return this.FromResult(result);
     }
 
+    [AuthorizeAny(Permissions.PUNCHITEM_CLEAR, Permissions.APPLICATION_TESTER)]
+    [HttpPost("{guid}/Reject")]
```
Comment (bratli) [inline src/Equinor.ProCoSys.Completion.WebApi/Controllers/PunchItems/PunchItemsController.cs:122]: Should these be POST? Why not PUT?

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+    }
+
+    public async Task<Result<string>> Handle(RejectPunchItemCommand request, CancellationToken cancellationToken)
+    {
+        var punchItem = await _punchItemRepository.GetByGuidAsync(request.PunchItemGuid);
+        if (punchItem == null)
+        {
+            throw new Exception($"Entity {nameof(PunchItem)} {request.PunchItemGuid} not found");
```
Comment (bratli) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs:39]: Should this be in the validator instead?

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandValidator.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+using System.Threading;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.Completion.Command.Validators.PunchItemValidators;
+using FluentValidation;
+
+namespace Equinor.ProCoSys.Completion.Command.PunchItemCommands.RejectPunchItem;
+
+public class RejectPunchItemCommandValidator : AbstractValidator<RejectPunchItemCommand>
+{
+    public RejectPunchItemCommandValidator(IPunchItemValidator punchItemValidator)
```
Comment (bratli) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandValidator.cs:11]: Do we not inherit the "HaveAValidRowVersion"-validation in validators for new API?

File: src/Equinor.ProCoSys.Completion.WebApi/Controllers/PunchItems/PunchItemsController.cs (hunk 1 above)
Comment (bratli) [reply src/Equinor.ProCoSys.Completion.WebApi/Controllers/PunchItems/PunchItemsController.cs:122]: Nvm, see that we use POST for similar endpoint in WebAPI

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs (hunk 2 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandHandler.cs:39]: it is ... this should never happen, more for "helping" setting up proper unit tests

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandValidator.cs (hunk 3 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/RejectPunchItem/RejectPunchItemCommandValidator.cs:11]: It is a input thing. The RowVersion is validated in Controller namespace in main project (as we did in preservation)
Doing input validation inside these business validators as we did in IPO, was a bad decision 
Was that the answer you searched for @bratli ?

**Review by bratli — APPROVED** [review]

**Pull request merged**